### PR TITLE
Release 0.5 preparations

### PR DESCRIPTION
**Pull request by str4d**
Part of #345.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]

File: zcash_client_backend/src/welding_rig.rs [hunk 1]
```diff
@@ -23,6 +23,8 @@ use crate::wallet::{AccountId, WalletShieldedOutput, WalletShieldedSpend, Wallet
 ///
 /// The given [`CommitmentTree`] and existing [`IncrementalWitness`]es are incremented
 /// with this output's commitment.
+///
+/// [`ScanningKey`]: zcash_client_backend::welding_rig::ScanningKey
```
Comment (str4d) [inline zcash_client_backend/src/welding_rig.rs:27]:
```suggestion
/// [`ScanningKey`]: crate::welding_rig::ScanningKey
```

File: zcash_client_backend/src/welding_rig.rs [hunk 2]
```diff
@@ -88,10 +90,23 @@ fn scan_output<P: consensus::Parameters, K: ScanningKey>(
...
+/// nullifier for the note can also be obtained.
+///
 /// [`CompactOutput`]: crate::proto::compact_formats::CompactOutput
+/// [`scan_block`]: crate::welding_rig::scan_block
 pub trait ScanningKey {
+    /// The type of nullifier extracted when a note is successfully
+    /// obtained by trial decryption.
     type Nf;
 
+    /// Attempt to decrypt a Sapling note and payment address
```
Comment (str4d) [inline zcash_client_backend/src/welding_rig.rs:108]: Per the Rust style guide:
```suggestion
    /// Attempts to decrypt a Sapling note and payment address
```

File: zcash_client_backend/src/welding_rig.rs [hunk 3]
```diff
@@ -101,9 +116,17 @@ pub trait ScanningKey {
         ct: &[u8],
     ) -> Option<(Note, PaymentAddress)>;
 
+    /// Produce the nullifier for the specified note and witness, if possible.
+    /// IVK-based implementations of this trait cannot successfully derive
```
Comment (str4d) [inline zcash_client_backend/src/welding_rig.rs:120]:
```suggestion
    /// Produces the nullifier for the specified note and witness, if possible.
    ///
    /// IVK-based implementations of this trait cannot successfully derive
```

File: zcash_client_backend/src/welding_rig.rs [hunk 4]
```diff
@@ -158,10 +185,11 @@ impl ScanningKey for SaplingIvk {
 /// [`ExtendedFullViewingKey`]: zcash_primitives::zip32::ExtendedFullViewingKey
 /// [`SaplingIvk`]: zcash_primitives::primitives::SaplingIvk
 /// [`CompactBlock`]: crate::proto::compact_formats::CompactBlock
-/// [`ScanningKey`]: self::ScanningKey
+/// [`ScanningKey`]: zcash_client_backend::welding_rig::ScanningKey
```
Comment (str4d) [inline zcash_client_backend/src/welding_rig.rs:188]:
```suggestion
/// [`ScanningKey`]: crate::welding_rig::ScanningKey
```

**Pull request merged**